cmd/build/v2/checker/checker_test.go

```diff
@@ -0,0 +1,378 @@
+			buildManifest: build.ManifestBuild{
+				"service1": &build.Info{Image: "test/service1"},
+			},
+			images:            []string{"service1"},
+			mockCached:        nil,
+			mockNotCached:     nil,
+			mockError:         errors.New("check strategy error"),
+			expectedCached:    nil,
+			expectedNotCached: nil,
+			expectedError:     true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Create mocks
+			mockCheckStrategy := &MockCheckStrategy{}
+			mockCheckStrategy.On("CheckServicesCache", mock.Anything, tt.manifestName, tt.buildManifest, tt.images).
+				Return(tt.mockCached, tt.mockNotCached, tt.mockError)
+
+			// Create checker with mock strategy
+			checker := &ImageCacheChecker{
+				checkStrategy: mockCheckStrategy,
+			}
+
+			// Execute
+			ctx := context.Background()
+			cached, notCached, err := checker.CheckImages(ctx, tt.manifestName, tt.buildManifest, tt.images)
+
+			// Verify results
+			if tt.expectedError {
+				assert.Error(t, err)
+				assert.Equal(t, tt.mockError, err)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tt.expectedCached, cached)
+				assert.Equal(t, tt.expectedNotCached, notCached)
+			}
+
+			// Verify mock was called
+			mockCheckStrategy.AssertExpectations(t)
+		})
+	}
+}
+
+func TestImageCacheChecker_GetImageDigestReferenceForServiceDeploy(t *testing.T) {
+	tests := []struct {
+		name           string
+		manifestName   string
+		service        string
+		buildInfo      *build.Info
+		mockDigest     string
+		mockError      error
+		expectedDigest string
+		expectedError  bool
+	}{
+		{
+			name:           "successful digest retrieval",
+			manifestName:   "test-manifest",
+			service:        "test-service",
+			buildInfo:      &build.Info{Image: "test/image"},
+			mockDigest:     "sha256:abc123",
+			mockError:      nil,
+			expectedDigest: "sha256:abc123",
+			expectedError:  false,
+		},
+		{
+			name:           "error from check strategy",
+			manifestName:   "test-manifest",
+			service:        "test-service",
+			buildInfo:      &build.Info{Image: "test/image"},
+			mockDigest:     "",
+			mockError:      errors.New("digest error"),
+			expectedDigest: "",
+			expectedError:  true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Create mocks
+			mockCheckStrategy := &MockCheckStrategy{}
+			mockCheckStrategy.On("GetImageDigestReferenceForServiceDeploy", tt.manifestName, tt.service, tt.buildInfo).
+				Return(tt.mockDigest, tt.mockError)
+
+			// Create checker with mock strategy
+			checker := &ImageCacheChecker{
+				checkStrategy: mockCheckStrategy,
+			}
+
+			// Execute
+			digest, err := checker.GetImageDigestReferenceForServiceDeploy(tt.manifestName, tt.service, tt.buildInfo)
+
+			// Verify results
+			if tt.expectedError {
+				assert.Error(t, err)
+				assert.Equal(t, tt.mockError, err)
+			} else {
+				assert.NoError(t, err)
+				assert.Equal(t, tt.expectedDigest, digest)
+			}
```

Comment:
We agreed on avoiding this kind of checks in test in favor of having more test functions (apply to other tests)